File: 백준/Bronze/16503. 괄호 없는 사칙연산/괄호 없는 사칙연산.py
import sys
input=lambda:sys.stdin.readline().rstrip()
k1,o1,k2,o2,k3=input().split()
def calculate(n1,o,n2):
    if o!='/':
        return str(eval(n1+o+n2))
    else:
        n1,n2=int(n1),int(n2)
        if n1<0 and n2>=0: return str((n1*-1//n2)*-1)
        elif n2<0 and n1>=0: return str((n1//(n2*-1))*-1)
        return str(n1//n2)
rlt1=int(calculate(k1,o1,calculate(k2,o2,k3)))
rlt2=int(calculate(calculate(k1,o1,k2),o2,k3))
print(min(rlt1,rlt2),max(rlt1,rlt2),sep="\n")

File: 백준/Bronze/16503. 괄호 없는 사칙연산/괄호 없는 사칙연산.c
#include <stdio.h>
int calculate(int n1, char o, int n2) {
    if (o != '/') {
        return o == '*' ? n1 * n2 : (o == '+' ? n1 + n2 : n1 - n2);
    } else {
        if (n1 < 0 && n2 >= 0)
            return ((n1 * -1) / n2) * -1;
        else if (n1 >= 0 && n2 < 0)
            return (n1 / (n2 * -1)) * -1;
        else
            return n1 / n2;
    }
}
int main(void) {
    int k1, k2, k3, rlt1, rlt2;
    char o1, o2;
    scanf("%d %c %d %c %d", &k1, &o1, &k2, &o2, &k3);
    rlt1 = calculate(calculate(k1, o1, k2), o2, k3);
    rlt2 = calculate(k1, o1, calculate(k2, o2, k3));
    printf("%d\n", rlt1 < rlt2 ? rlt1 : rlt2);
    printf("%d", rlt1 > rlt2 ? rlt1 : rlt2);
    return 0;
}